core/src/cluster_info.rs

```diff
@@ -533,6 +540,29 @@ struct ResponseScore {
     score: u64,             // Relative score of the response
 }
 
+// Retains only CRDS values associated with nodes with enough stake.
+// (some crds types are exempted)
+fn retain_staked(values: &mut Vec<CrdsValue>, stakes: &HashMap<Pubkey, u64>) {
+    values.retain(|value| {
+        match value.data {
+            CrdsData::ContactInfo(_) => true,
+            // May Impact new validators starting up without any stake yet.
+            CrdsData::Vote(_, _) => true,
+            // Unstaked nodes can still help repair.
+            CrdsData::EpochSlots(_, _) => true,
+            // Unstaked nodes can still serve snapshots.
+            CrdsData::SnapshotHashes(_) => true,
+            // Otherwise unstaked voting nodes will show up with no version in
+            // the various dashboards.
+            CrdsData::Version(_) => true,
+            _ => {
```

Comment:
Probably better to explicitly call out each CrdsData enum item here, to force the next person that adds a new CrdsData item to visit this function

Reply:
good point, done.